OpenRiaServices.DomainServices.Client/Test/Silverlight/Data/EntitySetTests.cs

```diff
@@ -246,6 +246,60 @@ public void ICVF_MemoryLeakTest()
             Assert.IsFalse(weakRef.IsAlive);
         }
 
+        [TestMethod]
+        public void InferredAddThenAttach_EntityCollection()
+        {
+            NorthwindEntityContainer ec = new NorthwindEntityContainer();
+
+            // track changes on products
+            int productAdded = 0;
+            int productRemoved = 0;
+            int productPropertyChanged = 0;
+            EntitySet<Product> productsSet = ec.GetEntitySet<Product>();
+            productsSet.EntityAdded += (obj, args) => productAdded += 1;
+            productsSet.EntityRemoved += (obj, args) => productRemoved += 1;
+            productsSet.PropertyChanged += (obj, args) => productPropertyChanged += 1;
```

Comment:
I think it would make sense to also check the inotifycollectionchanged implementation here as well.

Reply:
I just added that assertion now - I didn't realise at first that I could just cast the EntitySet to INotifyCollectionChanged. I forgot to mention earlier that I'm assuming the expected number of events in the first set of assertions are correct - they're not based on a deep understanding of what's supposed to be happening.